cmd/executor/thread-mate/main.go

```diff
@@ -0,0 +1,157 @@
+)
+
+const pluginName = "thread-mate"
+
+// version is set via ldflags by GoReleaser.
+var version = "dev"
+
+// ThreadMateExecutor implements the Botkube executor plugin interface.
+type ThreadMateExecutor struct {
+	once sync.Map
+}
+
+func NewThreadMateExecutor() *ThreadMateExecutor {
+	return &ThreadMateExecutor{}
+}
+
+// Metadata returns details about plugin.
+func (*ThreadMateExecutor) Metadata(context.Context) (api.MetadataOutput, error) {
+	return api.MetadataOutput{
+		Version:     version,
+		Description: "Streamlines managing assignment for incidents or user support",
+		JSONSchema: api.JSONSchema{
+			Value: thmate.JSONSchema,
+		},
+	}, nil
+}
+
+func (t *ThreadMateExecutor) init(cfg thmate.Config, kubeconfig []byte) (*thmate.ThreadMate, error) {
+	svc, ok := t.once.Load(cfg.RoundRobin.GroupName)
+	if ok {
+		return svc.(*thmate.ThreadMate), nil
+	}
+	kubeConfig, err := clientcmd.RESTConfigFromKubeConfig(kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("while reading kube config. %w", err)
+	}
+	k8sCli, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("while creating K8s clientset: %w", err)
+	}
+
+	cfgDumper := thmate.NewConfigMapDumper(k8sCli)
+
+	newSvc := thmate.New(cfg, cfgDumper)
+	newSvc.Start()
+
+	t.once.Store(cfg.RoundRobin.GroupName, newSvc)
+	return newSvc, nil
+}
+
+// Execute returns a given command as a response.
+func (t *ThreadMateExecutor) Execute(ctx context.Context, in executor.ExecuteInput) (executor.ExecuteOutput, error) {
+	if err := pluginx.ValidateKubeConfigProvided(pluginName, in.Context.KubeConfig); err != nil {
+		return executor.ExecuteOutput{}, err
+	}
+
+	var cmd thmate.Commands
+	err := pluginx.ParseCommand(pluginName, in.Command, &cmd)
+	switch {
+	case err == nil:
+	case errors.Is(err, arg.ErrHelp):
+		msg, _ := t.Help(ctx)
+		return executor.ExecuteOutput{
+			Message: msg,
+		}, nil
+	default:
+		return executor.ExecuteOutput{}, fmt.Errorf("while parsing input command: %w", err)
+	}
+
+	cfg, err := thmate.MergeConfigs(in.Configs)
+	if err != nil {
+		return executor.ExecuteOutput{}, fmt.Errorf("while merging configuration: %w", err)
+	}
+
+	svc, err := t.init(cfg, in.Context.KubeConfig)
+	if err != nil {
+		return executor.ExecuteOutput{}, fmt.Errorf("while initializing service: %w", err)
+	}
+
+	switch {
+	case cmd.Pick != nil:
+		msgs, err := svc.Pick(cmd.Pick, in.Context.Message)
+		if err != nil {
+			return executor.ExecuteOutput{}, err
+		}
+		if len(msgs) == 0 {
+			return executor.ExecuteOutput{
+				Message: api.Message{Type: api.SkipMessage},
+			}, nil
+		}
+		return executor.ExecuteOutput{
+			Messages: msgs,
+		}, nil
+	case cmd.Get != nil && cmd.Get.Activity != nil:
+		return executor.ExecuteOutput{
+			Message: svc.GetActivity(cmd.Get.Activity, in.Context.Message),
+		}, nil
+	case cmd.Resolve != nil:
+		return executor.ExecuteOutput{
+			Message: svc.Resolve(cmd.Resolve, in.Context.Message),
+		}, nil
+	case cmd.Takeover != nil:
```

Comment:
BTW later, in a follow-up, we could post a message in the thread that the "ticket" has been taken over by someone. Anyway, not needed for now 👍

Reply:
It would require a new feature on the socket slack side to support sth like:
```go
api.Message{
	ThreadResponseURL: "<>"
}
```
but I like the idea and it's totally doable 👍 sth to consider as the follow-up improvement 👍